sky/resources.py

```diff
@@ -869,6 +869,19 @@ def _try_validate_ports(self) -> None:
         if self.cloud is not None:
             self.cloud.check_features_are_supported(
                 self, {clouds.CloudImplementationFeatures.OPEN_PORTS})
+        else:
+            at_least_one_cloud_supports_ports = False
+            for cloud in global_user_state.get_enabled_clouds():
+                try:
+                    cloud.check_features_are_supported(
+                        self, {clouds.CloudImplementationFeatures.OPEN_PORTS})
+                    at_least_one_cloud_supports_ports = True
+                except exceptions.NotSupportedError:
+                    pass
+            if not at_least_one_cloud_supports_ports:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError('Cannot specify ports without enabling '
+                                     'a cloud that supports open ports.')
```

Comment:
The logic in execution.py should've caught this case?

Reply:
This is for better UX. The current master is already disabling the resources with ports if only cloud that does not support ports are enabled; but it will print some message like in #3125:

```bash
I 02-08 22:42:32 optimizer.py:1219] No resource satisfying <Cloud>(ports=['8080']) on RunPod.
sky.exceptions.ResourcesUnavailableError: Catalog does not contain any instances satisfying the request:
Task(run='conda activate vllm\...')
  resources: <Cloud>(ports=['8080']).
```